pipeline_dp/dp_computations.py

```diff
@@ -641,3 +643,57 @@ def create_additive_mechanism(
         return GaussianMechanism(spec.epsilon, spec.delta, sensitivities.l2)
 
     assert False, f"{spec.noise_kind} not supported."
+
+
+class ExponentialMechanism:
+    """Exponential mechanism that can be used to choose a parameter
+    from a set of possible parameters in a differentially private way.
+
+    https://en.wikipedia.org/wiki/Exponential_mechanism"""
```

Comment:
Please indicate limitations of this class in docstring: all computations are in memory, so the set from which we choose should fit in memory